src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/TagHelperBoundAttributeDescriptorExtensions.cs

```diff
@@ -1,29 +1,21 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#nullable disable
-
-using System;
-
 namespace Microsoft.AspNetCore.Razor.Language.Components;
 
 internal static class TagHelperBoundAttributeDescriptorExtensions
 {
+    private static bool IsTrue(this BoundAttributeDescriptor attribute, string key)
+        => attribute.Metadata.TryGetValue(key, out var value) && value == bool.TrueString;
```

Comment:
I didn't consider that, no. There are loads of these helper extension methods (mostly public) all over the place, so I decided to just follow that pattern. When we can finally change the public surface area of tag helper descriptors, I want to revisit all of this.

Reply:
Also, providing an interface wouldn't give us a single copy of the methods. We would have to copy the implementations to each builder type because there's no hierarchy relationship between them.